source/dub/internal/utils.d

```diff
@@ -30,28 +30,7 @@ version(DubUseCurl)
 	public import std.net.curl : HTTPStatusException;
 }
 
-
-private NativePath[] temporary_files;
-
-NativePath getTempDir()
-{
-	return NativePath(std.file.tempDir());
-}
-
-NativePath getTempFile(string prefix, string extension = null)
-{
-	import std.uuid : randomUUID;
-	import std.array: replace;
-
-	string fileName = prefix ~ "-" ~ randomUUID.toString() ~ extension;
-
-	if (extension !is null && extension == ".d")
-		fileName = fileName.replace("-", "_");
-
-	auto path = getTempDir() ~ fileName;
-	temporary_files ~= path;
-	return path;
-}
+public import dub.internal.temp_files;
```

Comment:
What was the cycle ?

Reply:
I don't have the output anymore, but it was about 6 imports deep, the deepest one before the cycle was dub.d which imported back on dub.internal.utils

Reply:
I'd like to know how `dub.internal.utils` can end up importing `dub.d`, as that shouldn't happen.

Reply:
it originates from the `import dub.package_ : packageInfoFiles;` inside utils.d

Reply:
Yeah that probably shouldn't be there...

Reply:
I think it's fine where that variable is defined, it's just the compiler being silly here and insisting on module constructor ordering.